### ISSUE TITLE
Cancel Action: allow Thing to reject

### ISSUE THREAD

**Issue by coderbyheart** [mozilla-iot/wot#55]
It may be that an action cannot be [canceled](https://github.com/mozilla-iot/wot/blob/9f6bab9897e9223b61aa4580c8418954eab2e3e9/index.html#L323) since it might be already "in flight" or the Thing is otherwise unable to cancel the action (e.g. because it currently does not have enough  resources).

In that case the Thing should be allowed to return the appropriate error code, which would be either `403 Forbidden` or `503 Service Unavailable`.

**Comment by benfrancis** [comment]
Yes, good idea. The API could respond with an error code like you say.

**Comment by benfrancis** [comment]
Note: currently our implementation sends a 404 if it fails to DELETE an action request, but we could try to be more specific https://github.com/mozilla-iot/gateway/blob/master/src/controllers/actions_controller.js#L88
